coding-agent/tests/coding_agent/environment/test_workspace_archive.py

```
@@ -0,0 +1,105 @@
+        _ = extract_workspace_archive_base64(
+            tmp_path / "target",
+            base64.b64encode(buffer.getvalue()).decode("ascii"),
+        )
+
+
+def test_workspace_archive_rejects_invalid_tar_without_clearing_target(tmp_path: Path) -> None:
+    target = tmp_path / "repo"
+    target.mkdir()
+    (target / "keep.txt").write_text("keep\n", encoding="utf-8")
+
+    with pytest.raises(ValueError, match="valid tar.gz"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)